http_logs/index.json

```diff
@@ -12,38 +12,118 @@
     "properties": {
       "@timestamp": {
         "format": "strict_date_optional_time||epoch_second",
-        "type": "date"
+        "type": "date",
+        "doc_values": {{ timestamp_doc_values | default(true) | tojson }},
+        "index": {{ timestamp_index | default(true) | tojson }}
       },
-      "clientip": {
-        "type": "ip"
-      },
-      "message": {
-        "type": "keyword",
-        "index": false,
-        "doc_values": false
-      },
-      "request": {
-        "type": "text",
-        "fields": {
-          "raw": {
-            "ignore_above": 256,
-            "type": "keyword"
+      {%- if runtime_grok is defined %}
+        "message": {
+          "type": "wildcard",
+          "index": true,
+          "doc_values": true
+        },
+        "extracted": {
+          "type": "runtime",
+          "grok": {
+            "field": "message",
+            "pattern": "%{IPORHOST:clientip} %{HTTPDUSER} %{USER} \\[%{TIMESTAMP_ISO8601}\\] \"(?:%{WORD} %{NOTSPACE:request}(?: HTTP/%{NUMBER})?|%{DATA})\" %{NUMBER:status:long} (?:%{NUMBER:size:long}|-)"
           }
+        },
+        "clientip": {
+          "type": "alias",
+          "path": "extracted.clientip"
+        },
+        "request": {
+          "type": "alias",
+          "path": "extracted.request"
+        },
+        "status": {
+          "type": "alias",
+          "path": "extracted.status"
+        },
+        "size": {
+          "type": "alias",
+          "path": "extracted.size"
         }
-      },
-      "status": {
-        "type": "integer"
-      },
-      "size": {
-        "type": "integer"
-      },
-      "geoip" : {
-        "properties" : {
-           "country_name": { "type": "keyword" },
-           "city_name": { "type": "keyword" },
-           "location" : { "type" : "geo_point" }
+      {%- elif runtime_script_grok is defined %}
```

Comment:
This implements using the a `script` to do grok-like-stuff. This is implemented in the 7.x branch but we're actively debating the API again.